on.js

```diff
@@ -581,11 +581,13 @@ define(["./has!dom-addeventlistener?:./aspect", "./_base/kernel", "./sniff"], fu
 						event.rotation = 0;
 						event.scale = 1;
 					}
-					//use event.changedTouches[0].pageX|pageY|screenX|screenY|clientX|clientY|target
-					var firstChangeTouch = event.changedTouches[0];
-					for(var i in firstChangeTouch){ // use for-in, we don't need to have dependency on dojo/_base/lang here
-						delete event[i]; // delete it first to make it mutable
-						event[i] = firstChangeTouch[i];
+					if (window.TouchEvent && originalEvent instanceof TouchEvent) {
```

Comment:
That is a good point @dylans! If someone were to emit a touch event programmatically (`on.emit(div, 'touchstart', { changedTouches: [{ pageX: 100 }] });`) that `has('touch-events')` test would not be sufficient and consulting the event instance for its type would be the only way to truly verify that it is a `TouchEvent`. Otherwise a `has` test would be preferable for sure.